utils/aur-check-update.py

```diff
@@ -85,43 +87,64 @@ def __str__(self) -> str:
         return f'{prefix}{self.pkgver}-{self.pkgrel}'
 
 
-async def main() -> int:
-    dry_run = '--dry' in sys.argv
-    os.chdir(sys.argv[1])
+async def with_sem(sem: asyncio.Semaphore, coro: Coroutine):
+    async with sem:
+        return await coro
+
+
+async def aur_check_update(item: Path, dry_run: bool) -> None:
     async with aiohttp.ClientSession(timeout=aiohttp.ClientTimeout(10)) as session:
-        for item in os.listdir():
-            if item.startswith('.'):
-                continue
-            if not os.path.isdir(item):
-                continue
-            os.chdir(item)
-            version = await Version.from_file('PKGBUILD')
-            async with session.get(f'https://aur.archlinux.org/cgit/aur.git/plain/PKGBUILD?h={item}') as response:
-                if response.status != 200:
-                    logging.info('%s not register in AUR', item)
-                    os.chdir('..')
-                    continue
-                new_version = Version.from_str(await response.text())
-            if new_version == version:
-                os.chdir('..')
-                continue
-            elif new_version < version:
-                logging.warning('Warning: %s is newer than AUR', version)
-                os.chdir('..')
-                continue
-            if '.git' in os.listdir():
-                if not dry_run:
-                    await asyncio.create_subprocess_exec('git', 'pull')
-                logging.info('Upgrade %s from %s to %s', item, version, new_version)
-            else:
-                logging.info('Found update %s(%s) (local: %s)', item, new_version, version)
-            os.chdir('..')
+        if item.name.startswith('.') or not item.is_dir():
+            return
+        pkgbuild_file = item.joinpath('PKGBUILD')
+        if not pkgbuild_file.exists():
+            logging.warning(f'PKGBUILD not in {str(item)}, SKIP!')
```

Comment:
Why use `str()` in  **formatted string literals**
```suggestion
            logging.warning(f'PKGBUILD not in {item}, SKIP!')
```